File: mavsdk-kotlin/src/commonMain/kotlin/io/mavsdk/kotlin/ConnectionResult.kt
package io.mavsdk.kotlin

/**
 * Connection result codes
 */
enum class ConnectionResult(val value: Int) {
    SUCCESS(0),
    TIMEOUT(1),
    SOCKET_ERROR(2),
    BIND_ERROR(3),
    SOCKET_CONNECTION_ERROR(4),
    CONNECTION_ERROR(5),
    NOT_IMPLEMENTED(6),
    SYSTEM_NOT_CONNECTED(7),
    SYSTEM_BUSY(8),
    COMMAND_DENIED(9),
    DESTINATION_IP_UNKNOWN(10),
    CONNECTIONS_EXHAUSTED(11),
    CONNECTION_URL_INVALID(12),
    BAUDRATE_UNKNOWN(13);

    companion object {
        fun fromValue(value: Int): ConnectionResult =
            entries.find { it.value == value } ?: CONNECTION_ERROR
    }
}

File: cpp/src/mavsdk/core/connection_result.cpp
#include "connection_result.h"
#include <ostream>

namespace mavsdk {

MAVSDK_PUBLIC std::string_view to_string(const ConnectionResult& result)
{
    switch (result) {
        case ConnectionResult::Success:
            return "Success";
        case ConnectionResult::Timeout:
            return "Timeout";
        case ConnectionResult::SocketError:
            return "Socket error";
        case ConnectionResult::BindError:
            return "Bind error";
        case ConnectionResult::SocketConnectionError:
            return "Socket connection error";
        case ConnectionResult::ConnectionError:
            return "Connection error";
        case ConnectionResult::NotImplemented:
            return "Not implemented";
        case ConnectionResult::SystemNotConnected:
            return "System not connected";
        case ConnectionResult::SystemBusy:
            return "System busy";
        case ConnectionResult::CommandDenied:
            return "Command denied";
        case ConnectionResult::DestinationIpUnknown:
            return "Destination IP unknown";
        case ConnectionResult::ConnectionsExhausted:
            return "Connections exhausted";
        case ConnectionResult::ConnectionUrlInvalid:
            return "Invalid connection URL";
        case ConnectionResult::BaudrateUnknown:
            return "Baudrate unknown";
        default:
            return "Unknown";
    }
}

MAVSDK_PUBLIC std::ostream& operator<<(std::ostream& str, const ConnectionResult& result)
{
    return str << to_string(result);
}

} // namespace mavsdk
